test/integration/commands/inspect/conftest.py

```diff
@@ -0,0 +1,328 @@
+    dummy_dir.mkdir(parents=True)
+    real_data.mkdir(parents=True)
+
+    with open(dummy_dir / 'dummy_file', 'w') as f:
+        f.write('dummy data')
+
+    (real_data / 'dummy_symlink').symlink_to(dummy_dir / 'dummy_file')
+
+
+# --- Internal utility functions.
+def _print_db_results(db_check_list, rows_as_dicts, run_tmpdir):
+    """
+    Print the database check results.
+    """
+    column_names = rows_as_dicts[0].keys() if rows_as_dicts else []
+    column_widths = [max(len(str(row[col])) for row in rows_as_dicts + [dict(zip(column_names, column_names))]) for col
+                     in column_names]
+    print(f"Experiment folder: {run_tmpdir}")
+    header = " | ".join(f"{name:<{width}}" for name, width in zip(column_names, column_widths))
+    print(f"\n{header}")
+    print("-" * len(header))
+    # Print the rows
+    for row_dict in rows_as_dicts:  # always print, for debug proposes
+        print(" | ".join(f"{str(row_dict[col]):<{width}}" for col, width in zip(column_names, column_widths)))
+    # Print the results
+    print("\nDatabase check results:")
+    print(f"JOB_DATA_EXIST: {db_check_list['JOB_DATA_EXIST']}")
+    print(f"AUTOSUBMIT_DB_EXIST: {db_check_list['AUTOSUBMIT_DB_EXIST']}")
+    print(f"JOB_DATA_ENTRIES_ARE_CORRECT: {db_check_list['JOB_DATA_ENTRIES']}")
+
+    for job_name in db_check_list["JOB_DATA_FIELDS"]:
+        for job_counter in db_check_list["JOB_DATA_FIELDS"][job_name]:
+            all_ok = True
+            for field in db_check_list["JOB_DATA_FIELDS"][job_name][job_counter]:
+                if field == "empty_fields":
+                    if len(db_check_list['JOB_DATA_FIELDS'][job_name][job_counter][field]) > 0:
+                        all_ok = False
+                        print(f"{field} assert FAILED")
+                else:
+                    if not db_check_list['JOB_DATA_FIELDS'][job_name][job_counter][field]:
+                        all_ok = False
+                        print(f"{field} assert FAILED")
+            if int(job_counter) > 0:
+                print(f"Job entry: {job_name} retrial: {job_counter} assert {str(all_ok).upper()}")
+            else:
+                print(f"Job entry: {job_name} assert {str(all_ok).upper()}")
+
+
+def run_in_thread(target: Callable[..., Any], *args, **kwargs) -> Thread:
+    """
+    Run the given target function in a separate thread.
+
+    :param target: The function to execute in the thread.
+    :type target: Callable[..., Any]
+    :param args: Positional arguments for the target function.
+    :type args: Any
+    :param kwargs: Keyword arguments for the target function.
+    :type kwargs: Any
+    :return: The started Thread object.
+    :rtype: Thread
+    """
+    thread = Thread(target=target, args=args, kwargs=kwargs)
+    thread.start()
+    return thread
+
+
+def _check_db_fields(run_tmpdir: Path, expected_entries, final_status, expid) -> dict[str, (bool, str)]:
```

Comment:
Just to sort out the mypy on the new files :raised_hands: 
```suggestion
def _check_db_fields(run_tmpdir: Path, expected_entries, final_status, expid) -> dict[str, tuple[bool, str]]:
```